### PR TITLE
Able to create P2V when just one IP is present - Fixed start&end time

### PR DESCRIPTION
**Pull request by rcesar999**
- Enhancement

## Proposed commit message
Two major changes are done to the logs-endace.flow-0.1.1-endace processors:
1. Added the capability to build the Pivot2Vision URL when the event received has only one IP (either source or destination). Two processsors were added, "Set IP Conversation if only source.ip is present" and "Set IP Conversation if only destination.ip is present"
2. Modified the start and end time. The search must be centered in the Timestamp, Starttime must be Timestamp-deltatime/2 and endtime must be Timesatmp+deltatime/2.

Notice that when an event has both IPs (source and destination), the ip_conv will be "ip_conv=srcip&dstip"
When only one IP is present in the event, ip_conv will be "ip=srcip" or "ip=dstip"

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

Thi PR is done as specified in 
https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md
in "Fork and clone the repository"

## How to test this PR locally

Flollow exemple from Elastic "Simulate pipeline API", open a Dev Tools console and enter:
POST /_ingest/pipeline/logs-endace.flow-0.1.1-endace/_simulate
{
  "docs":   [
    {
    "_source":         {
**enter test code here**
        } 
    }
  ]
}

The code comes from file "test-flow-events.json" For example, from line 121 to line 263.
To test the behaviour with just one IP event, just delete the field "source" and test, then put "source" back and delete field "destination" and test.
On the result window, look at the URL in the field "event"

## Related issues

None

## Screenshots

image

### CODE REVIEW

File: packages/endace/changelog.yml
Comment (jamesagarside) [inline packages/endace/changelog.yml:1]: This is change isnt needed as we havent released v0.1.0 yet. The overall Endace branch will act as the fist package release

File: packages/endace/manifest.yml
Comment (jamesagarside) [inline packages/endace/manifest.yml:1]: This also wont be needed as this PR will make up the v0.1.0 release

File: packages/endace/data_stream/flow/elasticsearch/ingest_pipeline/endace.yml
Comment (jamesagarside) [inline packages/endace/data_stream/flow/elasticsearch/ingest_pipeline/endace.yml:1]: @rcesar999 can you please update the tests to reflect these changes. These change will also need copying to the `packages/endace/data_stream/log/elasticsearch/ingest_pipeline/endace.yml` to ensure both methods of flow collection work

**Review by jamesagarside — CHANGES_REQUESTED** [review]
Great start, thank you. Few little comments. The main this is to ensure tests work as otherwise we wont be able to merge.

File: packages/endace/changelog.yml
Comment (rcesar999) [reply packages/endace/changelog.yml:1]: Moved it back to versio 0.1.0

File: packages/endace/manifest.yml
Comment (rcesar999) [reply packages/endace/manifest.yml:1]: Moved it back to versio 0.1.0

**Comment by rcesar999** [comment]
[roberto@elk-partner-8 pipeline]$ elastic-package test
2024/07/16 23:04:59  INFO New version is available - v0.101.2. Download from: https://github.com/elastic/elastic-package/releases/tag/v0.101.2
Run test suite for the package
Run asset tests for the package
2024/07/16 23:05:00  INFO Import ECS mappings into the built package (technical preview)
--- Test results for package: endace - START ---
????????????????????????????????????????????????????????????????????????????????????????????????????????????????
? PACKAGE ? DATA STREAM ? TEST TYPE ? TEST NAME                                        ? RESULT ? TIME ELAPSED ?
????????????????????????????????????????????????????????????????????????????????????????????????????????????????
? endace  ? flow        ? asset     ? index_template logs-endace.flow is loaded        ? PASS   ?      1.119µs ?
? endace  ? flow        ? asset     ? ingest_pipeline logs-endace.flow-0.1.0 is loaded ? PASS   ?        185ns ?
? endace  ? log         ? asset     ? index_template logs-endace.log is loaded         ? PASS   ?        172ns ?
? endace  ? log         ? asset     ? ingest_pipeline logs-endace.log-0.1.0 is loaded  ? PASS   ?        139ns ?
????????????????????????????????????????????????????????????????????????????????????????????????????????????????
--- Test results for package: endace - END   ---
Done
Run pipeline tests for the package
--- Test results for package: endace - START ---
????????????????????????????????????????????????????????????????????????????????????????????
? PACKAGE ? DATA STREAM ? TEST TYPE ? TEST NAME                    ? RESULT ? TIME ELAPSED ?
????????????????????????????????????????????????????????????????????????????????????????????
? endace  ? flow        ? pipeline  ? test-flow-events.json        ? PASS   ?  15.944467ms ?
? endace  ? flow        ? pipeline  ? (ingest pipeline warnings)   ? PASS   ? 281.745015ms ?
? endace  ? log         ? pipeline  ? test-netflow-log-events.json ? PASS   ?   15.99476ms ?
? endace  ? log         ? pipeline  ? (ingest pipeline warnings)   ? PASS   ? 290.786031ms ?
????????????????????????????????????????????????????????????????????????????????????????????
--- Test results for package: endace - END   ---
Done
Run static tests for the package
--- Test results for package: endace - START ---
????????????????????????????????????????????????????????????????????????????????????????
? PACKAGE ? DATA STREAM ? TEST TYPE ? TEST NAME                ? RESULT ? TIME ELAPSED ?
????????????????????????????????????????????????????????????????????????????????????????
? endace  ? flow        ? static    ? Verify sample_event.json ? PASS   ? 153.914408ms ?
? endace  ? log         ? static    ? Verify sample_event.json ? PASS   ? 306.634591ms ?
????????????????????????????????????????????????????????????????????????????????????????
--- Test results for package: endace - END   ---
Done
Run system tests for the package
2024/07/16 23:05:16  WARN setting up agent (technical preview)...
2024/07/16 23:05:25  INFO Import ECS mappings into the built package (technical preview)
2024/07/16 23:06:30  INFO Write container logs to file: /home/roberto/git/integrations/build/container-logs/endace-1721189190374847038.log
2024/07/16 23:06:31  INFO Write container logs to file: /home/roberto/git/integrations/build/container-logs/elastic-agent-1721189191064853618.log
2024/07/16 23:06:37  WARN setting up agent (technical preview)...
2024/07/16 23:06:45  INFO Import ECS mappings into the built package (technical preview)
2024/07/16 23:07:23  INFO Write container logs to file: /home/roberto/git/integrations/build/container-logs/endace-1721189243529998896.log
2024/07/16 23:07:24  INFO Write container logs to file: /home/roberto/git/integrations/build/container-logs/elastic-agent-1721189244043201659.log
2024/07/16 23:07:30  WARN setting up agent (technical preview)...
2024/07/16 23:07:39  INFO Import ECS mappings into the built package (technical preview)
2024/07/16 23:08:20  INFO Write container logs to file: /home/roberto/git/integrations/build/container-logs/endace-1721189300630466937.log
2024/07/16 23:08:21  INFO Write container logs to file: /home/roberto/git/integrations/build/container-logs/elastic-agent-1721189301180870174.log
2024/07/16 23:08:26  WARN setting up agent (technical preview)...
2024/07/16 23:08:35  INFO Import ECS mappings into the built package (technical preview)
2024/07/16 23:09:10  INFO Write container logs to file: /home/roberto/git/integrations/build/container-logs/endace-1721189350877550544.log
2024/07/16 23:09:11  INFO Write container logs to file: /home/roberto/git/integrations/build/container-logs/elastic-agent-1721189351404895716.log
2024/07/16 23:09:18  WARN setting up agent (technical preview)...
2024/07/16 23:09:27  INFO Import ECS mappings into the built package (technical preview)
2024/07/16 23:10:06  INFO Write container logs to file: /home/roberto/git/integrations/build/container-logs/netflow-log-netflow-1721189406959838366.log
2024/07/16 23:10:07  INFO Write container logs to file: /home/roberto/git/integrations/build/container-logs/elastic-agent-1721189407490705304.log
--- Test results for package: endace - START ---
???????????????????????????????????????????????????????????????????????????????????
? PACKAGE ? DATA STREAM ? TEST TYPE ? TEST NAME        ? RESULT ?    TIME ELAPSED ?
???????????????????????????????????????????????????????????????????????????????????
? endace  ? flow        ? system    ? http-get-2k-file ? PASS   ? 1m11.896524514s ?
? endace  ? flow        ? system    ? icmp-2-pings     ? PASS   ?   44.998294249s ?
? endace  ? flow        ? system    ? icmp4-ping       ? PASS   ?   47.987933313s ?
? endace  ? flow        ? system    ? icmp6-ping       ? PASS   ?   41.029689552s ?
? endace  ? log         ? system    ? netflow          ? PASS   ?   44.947407081s ?
???????????????????????????????????????????????????????????????????????????????????
--- Test results for package: endace - END   ---
Done
[roberto@elk-partner-8 pipeline]$

**Pull request merged**